test/mlir/krnl/krnl_bundle_lstm_memory_pool.mlir

```diff
@@ -0,0 +1,346 @@
+        %94 = addf %cst, %92 : f32
+        %95 = divf %93, %94 : f32
+        %96 = exp %90 : f32
+        %97 = subf %96, %cst : f32
+        %98 = addf %96, %cst : f32
+        %99 = divf %97, %98 : f32
+        %100 = cmpf "oge", %89, %cst_1 : f32
+        %101 = select %100, %95, %99 : f32
+        affine.store %101, %56[] : memref<f32>
+        %102 = affine.load %56[] : memref<f32>
+        %103 = mulf %83, %61 : f32
+        %104 = mulf %72, %102 : f32
+        %105 = addf %103, %104 : f32
+        affine.store %105, %7[%c0, %arg4, %arg5] : memref<1x?x3xf32>
+        %106 = affine.load %24[%arg4, %arg5] : memref<?x3xf32>
+        %107 = affine.load %28[%arg4, %arg5] : memref<?x3xf32>
+        %108 = addf %106, %107 : f32
+        %109 = alloc() : memref<4xi8>
+        %110 = "krnl.getref"(%109, %c0_i64) : (memref<4xi8>, i64) -> memref<f32>
+        affine.store %108, %110[] : memref<f32>
+        %111 = affine.load %110[] : memref<f32>
+        %112 = subf %cst_1, %111 : f32
+        %113 = exp %112 : f32
+        %114 = addf %cst, %113 : f32
+        %115 = divf %cst, %114 : f32
+        affine.store %115, %54[] : memref<f32>
+        %116 = affine.load %54[] : memref<f32>
+        %117 = alloc() : memref<4xi8>
+        %118 = "krnl.getref"(%117, %c0_i64) : (memref<4xi8>, i64) -> memref<f32>
+        affine.store %105, %118[] : memref<f32>
+        %119 = affine.load %118[] : memref<f32>
+        %120 = mulf %119, %cst_0 : f32
+        %121 = negf %120 : f32
+        %122 = exp %121 : f32
+        %123 = subf %cst, %122 : f32
+        %124 = addf %cst, %122 : f32
+        %125 = divf %123, %124 : f32
+        %126 = exp %120 : f32
+        %127 = subf %126, %cst : f32
+        %128 = addf %126, %cst : f32
+        %129 = divf %127, %128 : f32
+        %130 = cmpf "oge", %119, %cst_1 : f32
+        %131 = select %130, %125, %129 : f32
+        affine.store %131, %52[] : memref<f32>
+        %132 = affine.load %52[] : memref<f32>
+        %133 = mulf %116, %132 : f32
+        affine.store %133, %2[%c0, %arg4, %arg5] : memref<1x?x3xf32>
+        dealloc %117 : memref<4xi8>
+        dealloc %109 : memref<4xi8>
+        dealloc %87 : memref<4xi8>
+        dealloc %76 : memref<4xi8>
+        dealloc %65 : memref<4xi8>
+        dealloc %59 : memref<4xi8>
+        dealloc %57 : memref<4xi8>
+        dealloc %55 : memref<4xi8>
+        dealloc %53 : memref<4xi8>
+        dealloc %51 : memref<4xi8>
+      }
+      dealloc %45 : memref<?xi8>
+      dealloc %41 : memref<?xi8>
+      dealloc %36 : memref<?xi8>
+      dealloc %32 : memref<?xi8>
+      dealloc %27 : memref<?xi8>
+      dealloc %23 : memref<?xi8>
+      dealloc %18 : memref<?xi8>
+      dealloc %14 : memref<?xi8>
+    }
+    dealloc %6 : memref<?xi8>
+    return %2 : memref<1x?x3xf32>
+  }
+  "krnl.entry_point"() {func = @main_graph, numInputs = 3 : i32, numOutputs = 1 : i32} : () -> ()
+}
+
+// CHECK-LABEL: bundle_lstm_dyn_mem_pools
+
+// CHECK: [[C0:%.+]] = constant 0 : index
```

Comment:
I see a lot of operations that can go in parallel... It would be safer to use the CHECK-DAG

Reply:
Right, the only part which could be DAG'ed are the constants, otherwise for all the other operation the order is important.